edx_adapt/data/course_repository.py

```diff
@@ -407,41 +424,71 @@ def get_finished_users(self, course_id):
     def _get_problem(self, course_id, problem_name):
         problems = self.store.course_search(
                 course_id, 'problems', {'problem_name': problem_name}, 'problems', {'problem_name': problem_name}
-        )['problems']
+        ).get('problems')
         if problems:
             return problems[0]
         else:
             raise interface.DataException("Problem not found: {}".format(problem_name))
 
     def post_interaction(self, course_id, problem_name, user_id, correct, attempt, unix_seconds):
+        """
+        Store interaction notification in the database ..._log collection
+
+        :param course_id: course id
+        :param problem_name: name of the problem
+        :param user_id: student id
+        :param correct: boolean flag to mark answer is correct or not
+        :param attempt: number of attempts to answer the problem
+        :param unix_seconds: timestamp
+        """
         problem = self._get_problem(course_id, problem_name)
         if not problem:
-            # TODO(idegtiarov): I would avoid this checking in logging user operations to reduce request into database
             raise interface.DataException("Problem {} not in Course generic problems list".format(problem_name))
-        data = {'student_id': user_id, 'problem': problem, 'correct': correct, 'attempt': attempt,
-                'unix_s': unix_seconds, 'type': 'response',
-                'timestamp': datetime.datetime.fromtimestamp(unix_seconds).strftime('%Y-%m-%d %H:%M:%S')}
+        data = {
+            'student_id': user_id,
+            'problem': problem,
+            'correct': correct,
+            'attempt': attempt,
+            'unix_s': unix_seconds, 'type': 'response',
+            'timestamp': datetime.datetime.fromtimestamp(unix_seconds).strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
I usually do `from datetime import datetime` to avoid calling `datetime.datetime....`